src/Header/RequestParser.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "./OperateData.h"
+
+//used to make request string to OperateData(JSON like) format.
+class RequestParser
+{
+public:
+    //singleton code convention method.
+    static RequestParser* Get();
+    //give request string return header
+    static OperateData ParseHeader(std::string request);
+    //give request string return body / data
+    static OperateData ParseData(std::string request);
+private:
+    //singleton code convention method.
+    RequestParser();
+};
+
+namespace{
+    RequestParser* onlyRequestParser;
```

Comment:
This should go in the .cc file.

Reply:
Please add comment to note that it is the singleton instance.

Reply:
Note: It is recommended that you study why putting this in the header is a problem. If in doubt, ask me in class.

Reply:
Namespace removed